### PR TITLE
Deimos collate 1d

### PR DESCRIPTION
**Pull request by badpandabear**
Fix for Issue 1193:
Added exception handling around coadding so one failure doesn't stop the entire script.

Fix for Issue 1190:
Fixed get_object_based_metadata so that it no longer assumes that SpecObj data is shared for every row related to an object. 

Also made some improvements to the unit tests to increase code coverage.

### CODE REVIEW

**Comment by badpandabear** [comment]
Unit tests:
`================ 278 passed, 425 warnings in 774.11s (0:12:54) =================`

I'll have the dev suite run later tonight/tomorrow.

**Review by marlageha — APPROVED** [review]

**Comment by badpandabear** [comment]
Dev Suite Results:
```
--- PYPEIT DEVELOPMENT SUITE PASSED 112/112 TESTS (Masters ignored) ---
Testing Started at 2021-06-03T23:52:51.148802
Testing Completed at 2021-06-04T04:54:21.604378
Total Time: 5:01:30.455576
```

File: pypeit/scripts/collate_1d.py [hunk 1]
```diff
@@ -100,7 +101,7 @@ def get_metadata_by_id(header_keys, file_info):
 
     return ([data_row], file_info, filename)
 
-def get_object_based_metadata(object_header_keys, spec_obj_keys, source_header_keys, file_info):
+def get_object_based_metadata(object_header_keys, spec_obj_keys, file_info):
     """
     Gets the metadata from a SourceObject instance used for the by object
     portion of the archive. It is intended to be called by a
```
Comment (debora-pe) [inline pypeit/scripts/collate_1d.py]: Is this docstring outdated? It does not seem you have only one Argument.

**Review by debora-pe — APPROVED** [review]
One minor comment. Looks good!

**Pull request merged**